src/stream/src/executor/aggregation/minput.rs

```diff
@@ -102,9 +102,16 @@ impl MaterializedInputState {
                     .unzip()
             };
 
-        let pk_len = pk_indices.len();
-        order_col_indices.extend(pk_indices.iter());
-        order_types.extend(itertools::repeat_n(OrderType::ascending(), pk_len));
+        if agg_call.distinct {
+            if !order_col_indices.contains(&agg_call.args.val_indices()[0]) {
+                order_col_indices.push(agg_call.args.val_indices()[0]);
+                order_types.push(OrderType::ascending());
+            }
```

Comment:
Don't totally get the logic here, could you add some comments?

Reply:
This is corresponding to `infer_stream_agg_state` in optimizer. The logic is that, for distinct agg, we can't use `input pk` as extra keys to ensure uniqueness because it may contains keys not in `distinct keys`, leading to inconsistent `insert` and `delete` for state rows.

I guess it may be better to pass these `order_col_indices` and `order_types` from frontend to executor, but it's OK for now to leave it here.

Reply:
Detailed explaination on https://github.com/risingwavelabs/risingwave/issues/12140#issuecomment-1776289808

Reply:
comments and planner tests are added now.

Reply:
>  This is corresponding to infer_stream_agg_state in optimizer. The logic is that, for distinct agg, we can't use input pk as extra keys to ensure uniqueness because it may contains keys not in distinct keys, leading to inconsistent insert and delete for state rows.
>
> I guess it may be better to pass these order_col_indices and order_types from frontend to executor, but it's OK for now to leave it here.

+1. It should be passed from frontend.

Opened an issue to track: https://github.com/risingwavelabs/risingwave/issues/13044